plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRESTCatalog.java

```diff
@@ -0,0 +1,364 @@
+import io.trino.plugin.iceberg.catalog.TrinoCatalog;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorViewDefinition;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.security.TrinoPrincipal;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SessionCatalog;
+import org.apache.iceberg.catalog.SessionCatalog.SessionContext;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.RESTException;
+import org.apache.iceberg.rest.RESTSessionCatalog;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.hive.HiveMetadata.TABLE_COMMENT;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_CATALOG_ERROR;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.lang.String.format;
+
+public class TrinoRESTCatalog
+        implements TrinoCatalog
+{
+    private final RESTSessionCatalog sessionCatalog;
+    private final CatalogName catalogName;
+    private final String trinoVersion;
+
+    private final ConcurrentHashMap<String, Table> tableCache = new ConcurrentHashMap<>();
+
+    public TrinoRESTCatalog(
+            RESTSessionCatalog sessionCatalog,
+            CatalogName catalogName,
+            String trinoVersion)
+    {
+        this.sessionCatalog = sessionCatalog;
+        this.catalogName = catalogName;
+        this.trinoVersion = trinoVersion;
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return sessionCatalog.listNamespaces(convert(session)).stream()
+                .map(Namespace::toString).collect(Collectors.toList());
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        sessionCatalog.dropNamespace(convert(session), Namespace.of(namespace));
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return new HashMap<>(sessionCatalog.loadNamespaceMetadata(convert(session), Namespace.of(namespace)));
```

Comment:
It would be better to use `ImmutableMap.copyOf` if `loadNamespaceMetadata` doesn't return null in the values?